src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs

```diff
@@ -1453,18 +1671,48 @@ private static bool EnsureNumberOfValues(int segmentNumberOfValues, ref SingleRe
                             var currentTime = originalBaseline.AddMilliseconds(currentTimestamp);
                             while (true)
                             {
-                                var compare = Compare(currentTime, currentValues, localStatus, current, nextSegmentBaseline, timeSeriesSegment);
+                                // first check if local/remote entry status is `Dead`
+                                if (timeSeriesSegment.FromReplication &&
+                                    currentTime == current?.Timestamp && 
+                                    (localStatus == TimeSeriesValuesSegment.Dead || current.Status == TimeSeriesValuesSegment.Dead))
+                                {
+                                    if (localStatus == TimeSeriesValuesSegment.Dead)
+                                        timeSeriesSegment.AddExistingValue(currentTime, updatedValues, currentTag.AsSpan(), ref splitSegment, localStatus);
+                                    else
+                                    {
+                                        segmentChanged = true;
+                                        timeSeriesSegment.AddNewValue(current, ref splitSegment);
+                                    }
+                                    reader.MoveNext();
+                                    current = reader.Current;
+                                    break;
+                                }
+
+                                var compare = Compare(currentTime, currentValues, currentTag, localStatus, current, nextSegmentBaseline, timeSeriesSegment);
+
                                 if (compare != CompareResult.Remote)
                                 {
-                                    timeSeriesSegment.AddExistingValue(currentTime, updatedValues, currentTag.AsSpan(), ref splitSegment, localStatus);
-                                    if (currentTime == current?.Timestamp)
+                                    // possible scenarios:
+                                    // 1 - update from replication - local values >= remote values --> local wins
+                                    // 2 - local timestamp < remote timestamp
+                                    // 3 - equal timestamps & local tag < remote tag
```

Comment:
Why the sort order of the tags matter here?

Reply:
We maintain sorted order inside a segment (by timestamp and then by tag) and also in each list of append operations because we want the merge in `SplitSegment` to be O(n) instead of O(n^m) in the worst case.

For example, for the following scenraio, without sorted tags:
SegmentA
```
{12:20:00, value: x , tag: "alice"}
{12:20:00, value: x , tag: "bob"}
...
{12:20:00, value: x , tag: "ayende"}
```

Operations:
```
Append {12:20:00, value: z , tag: "baz"}
Append {12:20:00, value: z , tag: "aba"}
```

we need to move over all segment values with timestamp = "12:20:00" to check if the "baz" tag already exists and do the same thing for "aba"

with sorted tags:
SegmentA:
```
{12:20:00, value: x , tag: "alice"}
{12:20:00, value: x , tag: "ayende"}
...
{12:20:00, value: x , tag: "bob"}
```

Sorted operations:
```
Append {12:20:00, value: z , tag: "aba"}
Append {12:20:00, value: z , tag: "baz"}
```
-->
SegmentA:
```
 {12:20:00, value: z , tag: "aba"} // "aba" < "alice" - append new value
{12:20:00, value: x , tag: "alice"}  // "baz" > "alice" - append existing value
{12:20:00, value: x , tag: "ayende"} "baz" > "ayende" - append existing value
...
{12:20:00, value: z , tag: "baz"} // "baz" < "bob" - append new value
{12:20:00, value: x , tag: "bob"}
```